### ISSUE TITLE
--parallel_bs skips chunks of data

### ISSUE THREAD

**Issue by DigilConfianz** [m-bain/whisperX#155]
--vad_filter --parallel_bs [int] helps speed up the transcription process, but I observe that some chunks of text are skipped when running in batch using --parallel_bs. For example below is a transcript when ran with --parallel_bs
```
[06:08.108 --> 06:15.608]  No, the Google reviews do not syndicate on these listings, unfortunately.
[06:19.713 --> 06:26.713]  I can't. I don't know off the top of my head, but I want to say probably around 30, I think.
[06:26.713 --> 06:32.713]  Maybe a dash under 30. That's when I checked out at 24, so I'm not sure where it's at now.
[06:32.713 --> 06:36.713]  Do you do anything specific? Do you do anything specific to get reviews?
[06:36.713 --> 06:41.713]  Yes. I actually work with another company that manages...
```
and when ran using only --vad_filter without --parallel_bs

```
[00:18.500 --> 00:26.000]  No, the Google reviews do not syndicate on these listings, unfortunately.
[00:26.000 --> 00:29.000]  Okay? How many reviews do you have on Google?
~~ Transcribing VAD chunk: (06:19.713 --> 06:47.827) ~~
[00:00.000 --> 00:07.000]  I can't. I don't know off the top of my head, but I want to say probably around 30, I think.
[00:07.000 --> 00:13.000]  Maybe a dash under 30. That's when I checked out at 24, so I'm not sure where it's at now.
[00:13.000 --> 00:17.000]  Do you do anything specific? Do you do anything specific to get reviews?
[00:17.000 --> 00:22.000]  Yes. I actually work with another company that manages...
[00:22.000 --> 00:28.000]  My page is Optimus, and some of you also help with some of my internet.
```

clearly some chunks of data are being omitted, especially towards the end of a chunk. When checking code I think it maybe related to some padding issue?
https://github.com/m-bain/whisperX/blob/1e7c2c337b032347fb7cd4ed16312354a948b61a/whisperx/transcribe.py#L388
`chunk = torch.nn.functional.pad(chunk, (0, max_length-chunk.shape[-1]))`

find the audio here 
[test.zip](https://github.com/m-bain/whisperX/files/11099928/test.zip)

run as:
```
whisperx test.wav --hf_token hf_MYTOKEN --diarize --min_speakers 2 --max_speakers 10 --model small --vad_filter --language en
whisperx test.wav --hf_token hf_MYTOKEN --diarize --min_speakers 2 --max_speakers 10 --model small --vad_filter --language en --parallel_bs 2
```

**Comment by DigilConfianz** [comment]
[test.zip](https://github.com/m-bain/whisperX/files/11097425/test.zip)

whisperx test.wav  --hf_token hf_MYTOKEN --diarize --min_speakers 2 --max_speakers 10 --model small --vad_filter --language en
whisperx test.wav  --hf_token hf_MYTOKEN --diarize --min_speakers 2 --max_speakers 10 --model small --vad_filter --language en --parallel_bs 2

**Comment by kurianbenoy-sentient** [comment]
@DigilConfianz I have been noticing 4-5% accuracy difference while using parallel_bs and without using parallel_bs

**Comment by DigilConfianz** [comment]
V3 branch supports batching of chunks and the issue seems fixed in v3 branch.
